### PR TITLE
fix: do not ignore project name override for container projects

### PR DESCRIPTION
**Pull request by karniwl**
- [x] Ready for review
- [ ] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### How should this be manually tested?
Run `snyk monitor --docker <image-name> --project-name=<override-name>`

#### Any background context you want to provide?

A recent change introduced https://github.com/snyk/snyk/commit/183242cabec47f5506ca48208eea4fe78161624f#diff-cab76b228f009ba2ffcded98078566baR36 mistakenly ignored completely the override project name option using flag `--project-name`  for container projects. This commit fixes it and checks if `--project-name` is used.

### CODE REVIEW

File: src/lib/container/index.ts [hunk 1]
```diff
@@ -12,18 +13,28 @@ export function getContainerTargetFile(
...
   }
 }
 
 export function getContainerProjectName(
   scannedProject: ScannedProject,
+  meta: MonitorMeta,
 ): string | undefined {
-  return scannedProject.meta?.imageName;
+  let name = scannedProject.meta?.imageName;
+  if (meta['project-name'] && meta['project-name'] !== '') {
```
Comment (aviadhahami) [inline src/lib/container/index.ts:36]: `if (meta['project-name']?.length)` will simplify this a bit :)

File: src/lib/container/index.ts [hunk 2]
```diff
@@ -12,18 +13,28 @@ export function getContainerTargetFile(
 
 export function getContainerName(
   scannedProject: ScannedProject,
+  meta: MonitorMeta,
 ): string | undefined {
+  let name = scannedProject.meta?.imageName;
```
Comment (aviadhahami) [inline src/lib/container/index.ts:18]: what if there's no `imageName` (for some reason)?
maybe we should have a default value for rainy days?

Comment (karniwl) [reply src/lib/container/index.ts:18]: @firecode9 wdyt?

Comment (karniwl) [reply src/lib/container/index.ts:18]: that's how we validate it's a container project https://github.com/snyk/snyk/commit/183242cabec47f5506ca48208eea4fe78161624f#diff-2efe993c8fa18a460f73eb44bf5517c8R3

Comment (firecode9) [reply src/lib/container/index.ts:18]: afaik there must be an image name

Comment (firecode9) [reply src/lib/container/index.ts:18]: > that's how we validate it's a container project [183242c#diff-2efe993c8fa18a460f73eb44bf5517c8R3](https://github.com/snyk/snyk/commit/183242cabec47f5506ca48208eea4fe78161624f#diff-2efe993c8fa18a460f73eb44bf5517c8R3)

yes, thats what I meant 👍

Comment (aviadhahami) [reply src/lib/container/index.ts:18]: An empty string bypass these :smiling_imp:

Comment (firecode9) [reply src/lib/container/index.ts:18]: but the image name is given as a cli argument, so it must not be empty

Comment (aviadhahami) [reply src/lib/container/index.ts:18]: what if we also introduce this change:
```ts
export function isContainer(scannedProject: ScannedProject): boolean {
  return scannedProject.meta?.imageName.length;
}
```

Comment (aviadhahami) [reply src/lib/container/index.ts:18]: @firecode9 fair enough;
But may become problematic when you'll introduce API access(not sure if even planned)

Comment (firecode9) [reply src/lib/container/index.ts:18]: > what if we also introduce this change:
> 
> ```ts
> export function isContainer(scannedProject: ScannedProject): boolean {
>   return scannedProject.meta?.imageName.length;
> }
> ```

I think it wont do any harm to add this check

Comment (aviadhahami) [reply src/lib/container/index.ts:18]: All yours ;)

Comment (karniwl) [reply src/lib/container/index.ts:18]: adding :)

**Comment by karniwl** [comment]
@aviadhahami sorry to ping you again before the tests pass, we missed fixing some tests so had to push again, so your review is needed once again

**Pull request merged**